src/main/java/io/cdap/plugin/http/source/common/pagination/JythonPythonExecutor.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.http.source.common.pagination;
+
+import com.google.common.base.Strings;
+import org.python.core.PyException;
+import org.python.core.PyObject;
+import org.python.core.PySyntaxError;
+import org.python.util.PythonInterpreter;
+
+import java.io.Closeable;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Map;
+
+/**
+ * Executes python method 'get_next_page_url' using Jython library
+ */
+public class JythonPythonExecutor implements Closeable {
+  private static final String NEXT_PAGE_FUNCTION_NAME = "get_next_page_url";
+
+  private final String script;
+  private PythonInterpreter interpreter;
+  private PythonAccessor pythonAccessor;
+
+  public JythonPythonExecutor(String script) {
+    this.script = script;
+  }
+
+  public void initialize() {
+    interpreter = new PythonInterpreter();
+
+    try {
+      interpreter.exec(script);
+    } catch (PySyntaxError e) {
+      if (Strings.isNullOrEmpty(e.getMessage())) {
+        throw new RuntimeException(
+          "Unknown syntax error occurred while interpreting python code. Please make sure syntax is correct.", e);
+      }
+      throw new RuntimeException(
+        String.format("Syntax error occurred while interpreting python code: '%s'" +
```

Comment:
Should not surround the exception message with text, as it makes it hard to distinguish between what comes from the exception and what comes from this line. Better to just have:

Syntax error occurred while interpreting python code: '%s'

Reply:
done